// @ng-bootstrap/ng-bootstrap/modal/modal-container.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
import * as import0 from '@ng-bootstrap/ng-bootstrap/modal/modal-container';
export var Wrapper_NgbModalContainer = (function () {
    function Wrapper_NgbModalContainer(p0, p1, p2, p3, p4) {
        this._changed = false;
        this.context = new import0.NgbModalContainer(p0, p1, p2, p3, p4);
    }
    Wrapper_NgbModalContainer.prototype.ngOnDetach = function (view, componentView, el) {
    };
    Wrapper_NgbModalContainer.prototype.ngOnDestroy = function () {
    };
    Wrapper_NgbModalContainer.prototype.ngDoCheck = function (view, el, throwOnChange) {
        var changed = this._changed;
        this._changed = false;
        return changed;
    };
    Wrapper_NgbModalContainer.prototype.checkHost = function (view, componentView, el, throwOnChange) {
    };
    Wrapper_NgbModalContainer.prototype.handleEvent = function (eventName, $event) {
        var result = true;
        return result;
    };
    Wrapper_NgbModalContainer.prototype.subscribe = function (view, _eventHandler) {
        this._eventHandler = _eventHandler;
    };
    return Wrapper_NgbModalContainer;
}());
