// @ng-bootstrap/ng-bootstrap/modal/modal-backdrop.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
import * as import0 from '@ng-bootstrap/ng-bootstrap/modal/modal-backdrop';
import * as import1 from '@angular/core/src/linker/view';
import * as import3 from '@angular/core/src/linker/view_utils';
import * as import4 from '@angular/core/src/metadata/view';
import * as import5 from '@angular/core/src/linker/view_type';
import * as import6 from '@angular/core/src/change_detection/change_detection';
import * as import7 from '@angular/core/src/linker/component_factory';
export var Wrapper_NgbModalBackdrop = (function () {
    function Wrapper_NgbModalBackdrop() {
        this._changed = false;
        this.context = new import0.NgbModalBackdrop();
    }
    Wrapper_NgbModalBackdrop.prototype.ngOnDetach = function (view, componentView, el) {
    };
    Wrapper_NgbModalBackdrop.prototype.ngOnDestroy = function () {
    };
    Wrapper_NgbModalBackdrop.prototype.ngDoCheck = function (view, el, throwOnChange) {
        var changed = this._changed;
        this._changed = false;
        return changed;
    };
    Wrapper_NgbModalBackdrop.prototype.checkHost = function (view, componentView, el, throwOnChange) {
    };
    Wrapper_NgbModalBackdrop.prototype.handleEvent = function (eventName, $event) {
        var result = true;
        return result;
    };
    Wrapper_NgbModalBackdrop.prototype.subscribe = function (view, _eventHandler) {
        this._eventHandler = _eventHandler;
    };
    return Wrapper_NgbModalBackdrop;
}());
var renderType_NgbModalBackdrop_Host = import3.createRenderComponentType('', 0, import4.ViewEncapsulation.None, [], {});
var View_NgbModalBackdrop_Host0 = (function (_super) {
    __extends(View_NgbModalBackdrop_Host0, _super);
    function View_NgbModalBackdrop_Host0(viewUtils, parentView, parentIndex, parentElement) {
        _super.call(this, View_NgbModalBackdrop_Host0, renderType_NgbModalBackdrop_Host, import5.ViewType.HOST, viewUtils, parentView, parentIndex, parentElement, import6.ChangeDetectorStatus.CheckAlways);
    }
    View_NgbModalBackdrop_Host0.prototype.createInternal = function (rootSelector) {
        this._el_0 = import3.selectOrCreateRenderHostElement(this.renderer, 'ngb-modal-backdrop', new import3.InlineArray2(2, 'class', 'modal-backdrop fade in'), rootSelector, null);
        this.compView_0 = new View_NgbModalBackdrop0(this.viewUtils, this, 0, this._el_0);
        this._NgbModalBackdrop_0_3 = new Wrapper_NgbModalBackdrop();
        this.compView_0.create(this._NgbModalBackdrop_0_3.context);
        this.init(this._el_0, (this.renderer.directRenderer ? null : [this._el_0]), null);
        return new import7.ComponentRef_(0, this, this._el_0, this._NgbModalBackdrop_0_3.context);
    };
    View_NgbModalBackdrop_Host0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import0.NgbModalBackdrop) && (0 === requestNodeIndex))) {
            return this._NgbModalBackdrop_0_3.context;
        }
        return notFoundResult;
    };
    View_NgbModalBackdrop_Host0.prototype.detectChangesInternal = function (throwOnChange) {
        this._NgbModalBackdrop_0_3.ngDoCheck(this, this._el_0, throwOnChange);
        this.compView_0.detectChanges(throwOnChange);
    };
    View_NgbModalBackdrop_Host0.prototype.destroyInternal = function () {
        this.compView_0.destroy();
    };
    View_NgbModalBackdrop_Host0.prototype.visitRootNodesInternal = function (cb, ctx) {
        cb(this._el_0, ctx);
    };
    return View_NgbModalBackdrop_Host0;
}(import1.AppView));
export var NgbModalBackdropNgFactory = new import7.ComponentFactory('ngb-modal-backdrop', View_NgbModalBackdrop_Host0, import0.NgbModalBackdrop);
var styles_NgbModalBackdrop = [];
var renderType_NgbModalBackdrop = import3.createRenderComponentType('', 0, import4.ViewEncapsulation.None, styles_NgbModalBackdrop, {});
export var View_NgbModalBackdrop0 = (function (_super) {
    __extends(View_NgbModalBackdrop0, _super);
    function View_NgbModalBackdrop0(viewUtils, parentView, parentIndex, parentElement) {
        _super.call(this, View_NgbModalBackdrop0, renderType_NgbModalBackdrop, import5.ViewType.COMPONENT, viewUtils, parentView, parentIndex, parentElement, import6.ChangeDetectorStatus.CheckAlways);
    }
    View_NgbModalBackdrop0.prototype.createInternal = function (rootSelector) {
        var parentRenderNode = this.renderer.createViewRoot(this.parentElement);
        this.init(null, (this.renderer.directRenderer ? null : []), null);
        return null;
    };
    return View_NgbModalBackdrop0;
}(import1.AppView));
