// @ng-bootstrap/ng-bootstrap/tooltip/tooltip.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
import * as import0 from '@ng-bootstrap/ng-bootstrap/tooltip/tooltip';
import * as import1 from '@angular/core/src/change_detection/change_detection';
import * as import2 from '@angular/core/src/linker/view';
import * as import3 from '@angular/core/src/linker/view_utils';
import * as import5 from '@angular/core/src/metadata/view';
import * as import6 from '@angular/core/src/linker/view_type';
import * as import7 from '@angular/core/src/linker/component_factory';
export var Wrapper_NgbTooltip = (function () {
    function Wrapper_NgbTooltip(p0, p1, p2, p3, p4, p5, p6) {
        this._changed = false;
        this.context = new import0.NgbTooltip(p0, p1, p2, p3, p4, p5, p6);
        this._expr_0 = import1.UNINITIALIZED;
        this._expr_1 = import1.UNINITIALIZED;
        this._expr_2 = import1.UNINITIALIZED;
        this._expr_3 = import1.UNINITIALIZED;
    }
    Wrapper_NgbTooltip.prototype.ngOnDetach = function (view, componentView, el) {
    };
    Wrapper_NgbTooltip.prototype.ngOnDestroy = function () {
        this.context.ngOnDestroy();
        (this.subscription0 && this.subscription0.unsubscribe());
        (this.subscription1 && this.subscription1.unsubscribe());
    };
    Wrapper_NgbTooltip.prototype.check_placement = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_0, currValue))) {
            this._changed = true;
            this.context.placement = currValue;
            this._expr_0 = currValue;
        }
    };
    Wrapper_NgbTooltip.prototype.check_triggers = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_1, currValue))) {
            this._changed = true;
            this.context.triggers = currValue;
            this._expr_1 = currValue;
        }
    };
    Wrapper_NgbTooltip.prototype.check_container = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_2, currValue))) {
            this._changed = true;
            this.context.container = currValue;
            this._expr_2 = currValue;
        }
    };
    Wrapper_NgbTooltip.prototype.check_ngbTooltip = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_3, currValue))) {
            this._changed = true;
            this.context.ngbTooltip = currValue;
            this._expr_3 = currValue;
        }
    };
    Wrapper_NgbTooltip.prototype.ngDoCheck = function (view, el, throwOnChange) {
        var changed = this._changed;
        this._changed = false;
        if (!throwOnChange) {
            if ((view.numberOfChecks === 0)) {
                this.context.ngOnInit();
            }
        }
        return changed;
    };
    Wrapper_NgbTooltip.prototype.checkHost = function (view, componentView, el, throwOnChange) {
    };
    Wrapper_NgbTooltip.prototype.handleEvent = function (eventName, $event) {
        var result = true;
        return result;
    };
    Wrapper_NgbTooltip.prototype.subscribe = function (view, _eventHandler, emit0, emit1) {
        this._eventHandler = _eventHandler;
        if (emit0) {
            (this.subscription0 = this.context.shown.subscribe(_eventHandler.bind(view, 'shown')));
        }
        if (emit1) {
            (this.subscription1 = this.context.hidden.subscribe(_eventHandler.bind(view, 'hidden')));
        }
    };
    return Wrapper_NgbTooltip;
}());
export var Wrapper_NgbTooltipWindow = (function () {
    function Wrapper_NgbTooltipWindow() {
        this._changed = false;
        this.context = new import0.NgbTooltipWindow();
        this._expr_0 = import1.UNINITIALIZED;
        this._expr_1 = import1.UNINITIALIZED;
    }
    Wrapper_NgbTooltipWindow.prototype.ngOnDetach = function (view, componentView, el) {
    };
    Wrapper_NgbTooltipWindow.prototype.ngOnDestroy = function () {
    };
    Wrapper_NgbTooltipWindow.prototype.check_placement = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_0, currValue))) {
            this._changed = true;
            this.context.placement = currValue;
            this._expr_0 = currValue;
        }
    };
    Wrapper_NgbTooltipWindow.prototype.ngDoCheck = function (view, el, throwOnChange) {
        var changed = this._changed;
        this._changed = false;
        return changed;
    };
    Wrapper_NgbTooltipWindow.prototype.checkHost = function (view, componentView, el, throwOnChange) {
        var currVal_1 = ('tooltip in tooltip-' + this.context.placement);
        if (import3.checkBinding(throwOnChange, this._expr_1, currVal_1)) {
            view.renderer.setElementProperty(el, 'className', currVal_1);
            this._expr_1 = currVal_1;
        }
    };
    Wrapper_NgbTooltipWindow.prototype.handleEvent = function (eventName, $event) {
        var result = true;
        return result;
    };
    Wrapper_NgbTooltipWindow.prototype.subscribe = function (view, _eventHandler) {
        this._eventHandler = _eventHandler;
    };
    return Wrapper_NgbTooltipWindow;
}());
var renderType_NgbTooltipWindow_Host = import3.createRenderComponentType('', 0, import5.ViewEncapsulation.None, [], {});
var View_NgbTooltipWindow_Host0 = (function (_super) {
    __extends(View_NgbTooltipWindow_Host0, _super);
    function View_NgbTooltipWindow_Host0(viewUtils, parentView, parentIndex, parentElement) {
        _super.call(this, View_NgbTooltipWindow_Host0, renderType_NgbTooltipWindow_Host, import6.ViewType.HOST, viewUtils, parentView, parentIndex, parentElement, import1.ChangeDetectorStatus.CheckAlways);
    }
    View_NgbTooltipWindow_Host0.prototype.createInternal = function (rootSelector) {
        this._el_0 = import3.selectOrCreateRenderHostElement(this.renderer, 'ngb-tooltip-window', new import3.InlineArray2(2, 'role', 'tooltip'), rootSelector, null);
        this.compView_0 = new View_NgbTooltipWindow0(this.viewUtils, this, 0, this._el_0);
        this._NgbTooltipWindow_0_3 = new Wrapper_NgbTooltipWindow();
        this.compView_0.create(this._NgbTooltipWindow_0_3.context);
        this.init(this._el_0, (this.renderer.directRenderer ? null : [this._el_0]), null);
        return new import7.ComponentRef_(0, this, this._el_0, this._NgbTooltipWindow_0_3.context);
    };
    View_NgbTooltipWindow_Host0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import0.NgbTooltipWindow) && (0 === requestNodeIndex))) {
            return this._NgbTooltipWindow_0_3.context;
        }
        return notFoundResult;
    };
    View_NgbTooltipWindow_Host0.prototype.detectChangesInternal = function (throwOnChange) {
        if (this._NgbTooltipWindow_0_3.ngDoCheck(this, this._el_0, throwOnChange)) {
            this.compView_0.markAsCheckOnce();
        }
        this._NgbTooltipWindow_0_3.checkHost(this, this.compView_0, this._el_0, throwOnChange);
        this.compView_0.detectChanges(throwOnChange);
    };
    View_NgbTooltipWindow_Host0.prototype.destroyInternal = function () {
        this.compView_0.destroy();
    };
    View_NgbTooltipWindow_Host0.prototype.visitRootNodesInternal = function (cb, ctx) {
        cb(this._el_0, ctx);
    };
    View_NgbTooltipWindow_Host0.prototype.visitProjectableNodesInternal = function (nodeIndex, ngContentIndex, cb, ctx) {
        if (((nodeIndex == 0) && (ngContentIndex == 0))) { }
    };
    return View_NgbTooltipWindow_Host0;
}(import2.AppView));
export var NgbTooltipWindowNgFactory = new import7.ComponentFactory('ngb-tooltip-window', View_NgbTooltipWindow_Host0, import0.NgbTooltipWindow);
var styles_NgbTooltipWindow = [];
var renderType_NgbTooltipWindow = import3.createRenderComponentType('', 1, import5.ViewEncapsulation.None, styles_NgbTooltipWindow, {});
export var View_NgbTooltipWindow0 = (function (_super) {
    __extends(View_NgbTooltipWindow0, _super);
    function View_NgbTooltipWindow0(viewUtils, parentView, parentIndex, parentElement) {
        _super.call(this, View_NgbTooltipWindow0, renderType_NgbTooltipWindow, import6.ViewType.COMPONENT, viewUtils, parentView, parentIndex, parentElement, import1.ChangeDetectorStatus.CheckOnce);
    }
    View_NgbTooltipWindow0.prototype.createInternal = function (rootSelector) {
        var parentRenderNode = this.renderer.createViewRoot(this.parentElement);
        this._text_0 = this.renderer.createText(parentRenderNode, '\n    ', null);
        this._el_1 = import3.createRenderElement(this.renderer, parentRenderNode, 'div', new import3.InlineArray2(2, 'class', 'tooltip-arrow'), null);
        this._text_2 = this.renderer.createText(parentRenderNode, '\n    ', null);
        this._el_3 = import3.createRenderElement(this.renderer, parentRenderNode, 'div', new import3.InlineArray2(2, 'class', 'tooltip-inner'), null);
        this.projectNodes(this._el_3, 0);
        this._text_4 = this.renderer.createText(parentRenderNode, '\n    ', null);
        this.init(null, (this.renderer.directRenderer ? null : [
            this._text_0,
            this._el_1,
            this._text_2,
            this._el_3,
            this._text_4
        ]), null);
        return null;
    };
    return View_NgbTooltipWindow0;
}(import2.AppView));
