// ng2-file-upload/file-upload/file-drop.directive.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
import * as import0 from 'ng2-file-upload/file-upload/file-drop.directive';
import * as import1 from '@angular/core/src/change_detection/change_detection';
import * as import3 from '@angular/core/src/linker/view_utils';
export var Wrapper_FileDropDirective = (function () {
    function Wrapper_FileDropDirective(p0) {
        this._changed = false;
        this.context = new import0.FileDropDirective(p0);
        this._expr_0 = import1.UNINITIALIZED;
    }
    Wrapper_FileDropDirective.prototype.ngOnDetach = function (view, componentView, el) {
    };
    Wrapper_FileDropDirective.prototype.ngOnDestroy = function () {
        (this.subscription0 && this.subscription0.unsubscribe());
        (this.subscription1 && this.subscription1.unsubscribe());
    };
    Wrapper_FileDropDirective.prototype.check_uploader = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_0, currValue))) {
            this._changed = true;
            this.context.uploader = currValue;
            this._expr_0 = currValue;
        }
    };
    Wrapper_FileDropDirective.prototype.ngDoCheck = function (view, el, throwOnChange) {
        var changed = this._changed;
        this._changed = false;
        return changed;
    };
    Wrapper_FileDropDirective.prototype.checkHost = function (view, componentView, el, throwOnChange) {
    };
    Wrapper_FileDropDirective.prototype.handleEvent = function (eventName, $event) {
        var result = true;
        if ((eventName == 'drop')) {
            var pd_sub_0 = (this.context.onDrop($event) !== false);
            result = (pd_sub_0 && result);
        }
        if ((eventName == 'dragover')) {
            var pd_sub_1 = (this.context.onDragOver($event) !== false);
            result = (pd_sub_1 && result);
        }
        if ((eventName == 'dragleave')) {
            var pd_sub_2 = (this.context.onDragLeave($event) !== false);
            result = (pd_sub_2 && result);
        }
        return result;
    };
    Wrapper_FileDropDirective.prototype.subscribe = function (view, _eventHandler, emit0, emit1) {
        this._eventHandler = _eventHandler;
        if (emit0) {
            (this.subscription0 = this.context.fileOver.subscribe(_eventHandler.bind(view, 'fileOver')));
        }
        if (emit1) {
            (this.subscription1 = this.context.onFileDrop.subscribe(_eventHandler.bind(view, 'onFileDrop')));
        }
    };
    return Wrapper_FileDropDirective;
}());
