// @ng-bootstrap/ng-bootstrap/typeahead/highlight.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
import * as import0 from '@ng-bootstrap/ng-bootstrap/typeahead/highlight';
import * as import1 from '@angular/core/src/change_detection/change_detection';
import * as import2 from '@angular/core/src/linker/view';
import * as import3 from '@angular/core/src/linker/view_utils';
import * as import5 from '@angular/core/src/metadata/view';
import * as import6 from '@angular/core/src/linker/view_type';
import * as import7 from '@angular/core/src/linker/component_factory';
import * as import8 from '@angular/core/src/linker/view_container';
import * as import9 from '../../../@angular/common/src/directives/ng_for.ngfactory';
import * as import10 from '@angular/core/src/linker/template_ref';
import * as import11 from '@angular/core/src/change_detection/differs/iterable_differs';
import * as import12 from '@angular/common/src/directives/ng_for';
import * as import13 from '../../../@angular/common/src/directives/ng_if.ngfactory';
import * as import14 from '@angular/common/src/directives/ng_if';
export var Wrapper_NgbHighlight = (function () {
    function Wrapper_NgbHighlight() {
        this._changed = false;
        this._changes = {};
        this.context = new import0.NgbHighlight();
        this._expr_0 = import1.UNINITIALIZED;
        this._expr_1 = import1.UNINITIALIZED;
        this._expr_2 = import1.UNINITIALIZED;
    }
    Wrapper_NgbHighlight.prototype.ngOnDetach = function (view, componentView, el) {
    };
    Wrapper_NgbHighlight.prototype.ngOnDestroy = function () {
    };
    Wrapper_NgbHighlight.prototype.check_highlightClass = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_0, currValue))) {
            this._changed = true;
            this.context.highlightClass = currValue;
            this._changes['highlightClass'] = new import1.SimpleChange(this._expr_0, currValue);
            this._expr_0 = currValue;
        }
    };
    Wrapper_NgbHighlight.prototype.check_result = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_1, currValue))) {
            this._changed = true;
            this.context.result = currValue;
            this._changes['result'] = new import1.SimpleChange(this._expr_1, currValue);
            this._expr_1 = currValue;
        }
    };
    Wrapper_NgbHighlight.prototype.check_term = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_2, currValue))) {
            this._changed = true;
            this.context.term = currValue;
            this._changes['term'] = new import1.SimpleChange(this._expr_2, currValue);
            this._expr_2 = currValue;
        }
    };
    Wrapper_NgbHighlight.prototype.ngDoCheck = function (view, el, throwOnChange) {
        var changed = this._changed;
        this._changed = false;
        if (!throwOnChange) {
            if (changed) {
                this.context.ngOnChanges(this._changes);
                this._changes = {};
            }
        }
        return changed;
    };
    Wrapper_NgbHighlight.prototype.checkHost = function (view, componentView, el, throwOnChange) {
    };
    Wrapper_NgbHighlight.prototype.handleEvent = function (eventName, $event) {
        var result = true;
        return result;
    };
    Wrapper_NgbHighlight.prototype.subscribe = function (view, _eventHandler) {
        this._eventHandler = _eventHandler;
    };
    return Wrapper_NgbHighlight;
}());
var renderType_NgbHighlight_Host = import3.createRenderComponentType('', 0, import5.ViewEncapsulation.None, [], {});
var View_NgbHighlight_Host0 = (function (_super) {
    __extends(View_NgbHighlight_Host0, _super);
    function View_NgbHighlight_Host0(viewUtils, parentView, parentIndex, parentElement) {
        _super.call(this, View_NgbHighlight_Host0, renderType_NgbHighlight_Host, import6.ViewType.HOST, viewUtils, parentView, parentIndex, parentElement, import1.ChangeDetectorStatus.CheckAlways);
    }
    View_NgbHighlight_Host0.prototype.createInternal = function (rootSelector) {
        this._el_0 = import3.selectOrCreateRenderHostElement(this.renderer, 'ngb-highlight', import3.EMPTY_INLINE_ARRAY, rootSelector, null);
        this.compView_0 = new View_NgbHighlight0(this.viewUtils, this, 0, this._el_0);
        this._NgbHighlight_0_3 = new Wrapper_NgbHighlight();
        this.compView_0.create(this._NgbHighlight_0_3.context);
        this.init(this._el_0, (this.renderer.directRenderer ? null : [this._el_0]), null);
        return new import7.ComponentRef_(0, this, this._el_0, this._NgbHighlight_0_3.context);
    };
    View_NgbHighlight_Host0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import0.NgbHighlight) && (0 === requestNodeIndex))) {
            return this._NgbHighlight_0_3.context;
        }
        return notFoundResult;
    };
    View_NgbHighlight_Host0.prototype.detectChangesInternal = function (throwOnChange) {
        if (this._NgbHighlight_0_3.ngDoCheck(this, this._el_0, throwOnChange)) {
            this.compView_0.markAsCheckOnce();
        }
        this.compView_0.detectChanges(throwOnChange);
    };
    View_NgbHighlight_Host0.prototype.destroyInternal = function () {
        this.compView_0.destroy();
    };
    View_NgbHighlight_Host0.prototype.visitRootNodesInternal = function (cb, ctx) {
        cb(this._el_0, ctx);
    };
    return View_NgbHighlight_Host0;
}(import2.AppView));
export var NgbHighlightNgFactory = new import7.ComponentFactory('ngb-highlight', View_NgbHighlight_Host0, import0.NgbHighlight);
var styles_NgbHighlight = ['.ngb-highlight[_ngcontent-%COMP%] {\n      font-weight: bold;\n    }'];
var renderType_NgbHighlight = import3.createRenderComponentType('', 0, import5.ViewEncapsulation.Emulated, styles_NgbHighlight, {});
export var View_NgbHighlight0 = (function (_super) {
    __extends(View_NgbHighlight0, _super);
    function View_NgbHighlight0(viewUtils, parentView, parentIndex, parentElement) {
        _super.call(this, View_NgbHighlight0, renderType_NgbHighlight, import6.ViewType.COMPONENT, viewUtils, parentView, parentIndex, parentElement, import1.ChangeDetectorStatus.CheckOnce);
    }
    View_NgbHighlight0.prototype.createInternal = function (rootSelector) {
        var parentRenderNode = this.renderer.createViewRoot(this.parentElement);
        this._anchor_0 = this.renderer.createTemplateAnchor(parentRenderNode, null);
        this._vc_0 = new import8.ViewContainer(0, null, this, this._anchor_0);
        this._TemplateRef_0_5 = new import10.TemplateRef_(this, 0, this._anchor_0);
        this._NgFor_0_6 = new import9.Wrapper_NgFor(this._vc_0.vcRef, this._TemplateRef_0_5, this.parentView.injectorGet(import11.IterableDiffers, this.parentIndex), this.ref);
        this.init(null, (this.renderer.directRenderer ? null : [this._anchor_0]), null);
        return null;
    };
    View_NgbHighlight0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import10.TemplateRef) && (0 === requestNodeIndex))) {
            return this._TemplateRef_0_5;
        }
        if (((token === import12.NgFor) && (0 === requestNodeIndex))) {
            return this._NgFor_0_6.context;
        }
        return notFoundResult;
    };
    View_NgbHighlight0.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_0_0_0 = this.context.parts;
        this._NgFor_0_6.check_ngForOf(currVal_0_0_0, throwOnChange, false);
        this._NgFor_0_6.ngDoCheck(this, this._anchor_0, throwOnChange);
        this._vc_0.detectChangesInNestedViews(throwOnChange);
    };
    View_NgbHighlight0.prototype.destroyInternal = function () {
        this._vc_0.destroyNestedViews();
    };
    View_NgbHighlight0.prototype.createEmbeddedViewInternal = function (nodeIndex) {
        if ((nodeIndex == 0)) {
            return new View_NgbHighlight1(this.viewUtils, this, 0, this._anchor_0, this._vc_0);
        }
        return null;
    };
    return View_NgbHighlight0;
}(import2.AppView));
var View_NgbHighlight1 = (function (_super) {
    __extends(View_NgbHighlight1, _super);
    function View_NgbHighlight1(viewUtils, parentView, parentIndex, parentElement, declaredViewContainer) {
        _super.call(this, View_NgbHighlight1, renderType_NgbHighlight, import6.ViewType.EMBEDDED, viewUtils, parentView, parentIndex, parentElement, import1.ChangeDetectorStatus.CheckAlways, declaredViewContainer);
    }
    View_NgbHighlight1.prototype.createInternal = function (rootSelector) {
        this._anchor_0 = this.renderer.createTemplateAnchor(null, null);
        this._vc_0 = new import8.ViewContainer(0, null, this, this._anchor_0);
        this._TemplateRef_0_5 = new import10.TemplateRef_(this, 0, this._anchor_0);
        this._NgIf_0_6 = new import13.Wrapper_NgIf(this._vc_0.vcRef, this._TemplateRef_0_5);
        this._anchor_1 = this.renderer.createTemplateAnchor(null, null);
        this._vc_1 = new import8.ViewContainer(1, null, this, this._anchor_1);
        this._TemplateRef_1_5 = new import10.TemplateRef_(this, 1, this._anchor_1);
        this._NgIf_1_6 = new import13.Wrapper_NgIf(this._vc_1.vcRef, this._TemplateRef_1_5);
        this._el_2 = this.renderer.createTemplateAnchor(null, null);
        this.init(this._el_2, (this.renderer.directRenderer ? null : [
            this._anchor_0,
            this._anchor_1
        ]), null);
        return null;
    };
    View_NgbHighlight1.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import10.TemplateRef) && (0 === requestNodeIndex))) {
            return this._TemplateRef_0_5;
        }
        if (((token === import14.NgIf) && (0 === requestNodeIndex))) {
            return this._NgIf_0_6.context;
        }
        if (((token === import10.TemplateRef) && (1 === requestNodeIndex))) {
            return this._TemplateRef_1_5;
        }
        if (((token === import14.NgIf) && (1 === requestNodeIndex))) {
            return this._NgIf_1_6.context;
        }
        return notFoundResult;
    };
    View_NgbHighlight1.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_0_0_0 = this.context.odd;
        this._NgIf_0_6.check_ngIf(currVal_0_0_0, throwOnChange, false);
        this._NgIf_0_6.ngDoCheck(this, this._anchor_0, throwOnChange);
        var currVal_1_0_0 = !this.context.odd;
        this._NgIf_1_6.check_ngIf(currVal_1_0_0, throwOnChange, false);
        this._NgIf_1_6.ngDoCheck(this, this._anchor_1, throwOnChange);
        this._vc_0.detectChangesInNestedViews(throwOnChange);
        this._vc_1.detectChangesInNestedViews(throwOnChange);
    };
    View_NgbHighlight1.prototype.destroyInternal = function () {
        this._vc_0.destroyNestedViews();
        this._vc_1.destroyNestedViews();
    };
    View_NgbHighlight1.prototype.visitRootNodesInternal = function (cb, ctx) {
        cb(this._vc_0.nativeElement, ctx);
        this._vc_0.visitNestedViewRootNodes(cb, ctx);
        cb(this._vc_1.nativeElement, ctx);
        this._vc_1.visitNestedViewRootNodes(cb, ctx);
        cb(this._el_2, ctx);
    };
    View_NgbHighlight1.prototype.createEmbeddedViewInternal = function (nodeIndex) {
        if ((nodeIndex == 0)) {
            return new View_NgbHighlight2(this.viewUtils, this, 0, this._anchor_0, this._vc_0);
        }
        if ((nodeIndex == 1)) {
            return new View_NgbHighlight3(this.viewUtils, this, 1, this._anchor_1, this._vc_1);
        }
        return null;
    };
    return View_NgbHighlight1;
}(import2.AppView));
var View_NgbHighlight2 = (function (_super) {
    __extends(View_NgbHighlight2, _super);
    function View_NgbHighlight2(viewUtils, parentView, parentIndex, parentElement, declaredViewContainer) {
        _super.call(this, View_NgbHighlight2, renderType_NgbHighlight, import6.ViewType.EMBEDDED, viewUtils, parentView, parentIndex, parentElement, import1.ChangeDetectorStatus.CheckAlways, declaredViewContainer);
        this._expr_2 = import1.UNINITIALIZED;
        this._expr_3 = import1.UNINITIALIZED;
    }
    View_NgbHighlight2.prototype.createInternal = function (rootSelector) {
        this._el_0 = import3.createRenderElement(this.renderer, null, 'span', import3.EMPTY_INLINE_ARRAY, null);
        this._text_1 = this.renderer.createText(this._el_0, '', null);
        this.init(this._el_0, (this.renderer.directRenderer ? null : [
            this._el_0,
            this._text_1
        ]), null);
        return null;
    };
    View_NgbHighlight2.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_2 = import3.inlineInterpolate(1, '', this.parentView.parentView.context.highlightClass, '');
        if (import3.checkBinding(throwOnChange, this._expr_2, currVal_2)) {
            this.renderer.setElementProperty(this._el_0, 'className', currVal_2);
            this._expr_2 = currVal_2;
        }
        var currVal_3 = import3.inlineInterpolate(1, '', this.parentView.context.$implicit, '');
        if (import3.checkBinding(throwOnChange, this._expr_3, currVal_3)) {
            this.renderer.setText(this._text_1, currVal_3);
            this._expr_3 = currVal_3;
        }
    };
    View_NgbHighlight2.prototype.visitRootNodesInternal = function (cb, ctx) {
        cb(this._el_0, ctx);
    };
    return View_NgbHighlight2;
}(import2.AppView));
var View_NgbHighlight3 = (function (_super) {
    __extends(View_NgbHighlight3, _super);
    function View_NgbHighlight3(viewUtils, parentView, parentIndex, parentElement, declaredViewContainer) {
        _super.call(this, View_NgbHighlight3, renderType_NgbHighlight, import6.ViewType.EMBEDDED, viewUtils, parentView, parentIndex, parentElement, import1.ChangeDetectorStatus.CheckAlways, declaredViewContainer);
        this._expr_1 = import1.UNINITIALIZED;
    }
    View_NgbHighlight3.prototype.createInternal = function (rootSelector) {
        this._text_0 = this.renderer.createText(null, '', null);
        this.init(this._text_0, (this.renderer.directRenderer ? null : [this._text_0]), null);
        return null;
    };
    View_NgbHighlight3.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_1 = import3.inlineInterpolate(1, '', this.parentView.context.$implicit, '');
        if (import3.checkBinding(throwOnChange, this._expr_1, currVal_1)) {
            this.renderer.setText(this._text_0, currVal_1);
            this._expr_1 = currVal_1;
        }
    };
    View_NgbHighlight3.prototype.visitRootNodesInternal = function (cb, ctx) {
        cb(this._text_0, ctx);
    };
    return View_NgbHighlight3;
}(import2.AppView));
