// @ng-bootstrap/ng-bootstrap/datepicker/datepicker-navigation.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
import * as import0 from '@ng-bootstrap/ng-bootstrap/datepicker/datepicker-navigation';
import * as import1 from '@angular/core/src/change_detection/change_detection';
import * as import2 from '@angular/core/src/linker/view';
import * as import3 from '@angular/core/src/linker/view_utils';
import * as import5 from '@angular/core/src/metadata/view';
import * as import6 from '@angular/core/src/linker/view_type';
import * as import7 from '@angular/core/src/linker/component_factory';
import * as import8 from '@ng-bootstrap/ng-bootstrap/datepicker/datepicker-i18n';
import * as import9 from '@ng-bootstrap/ng-bootstrap/datepicker/ngb-calendar';
import * as import10 from '@angular/core/src/linker/view_container';
import * as import11 from '../../../@angular/common/src/directives/ng_if.ngfactory';
import * as import12 from '@angular/core/src/linker/template_ref';
import * as import13 from '@angular/common/src/directives/ng_if';
import * as import14 from '@ng-bootstrap/ng-bootstrap/datepicker/datepicker-navigation-select';
import * as import15 from './datepicker-navigation-select.ngfactory';
export var Wrapper_NgbDatepickerNavigation = (function () {
    function Wrapper_NgbDatepickerNavigation(p0, p1) {
        this._changed = false;
        this.context = new import0.NgbDatepickerNavigation(p0, p1);
        this._expr_0 = import1.UNINITIALIZED;
        this._expr_1 = import1.UNINITIALIZED;
        this._expr_2 = import1.UNINITIALIZED;
        this._expr_3 = import1.UNINITIALIZED;
        this._expr_4 = import1.UNINITIALIZED;
        this._expr_5 = import1.UNINITIALIZED;
    }
    Wrapper_NgbDatepickerNavigation.prototype.ngOnDetach = function (view, componentView, el) {
    };
    Wrapper_NgbDatepickerNavigation.prototype.ngOnDestroy = function () {
        (this.subscription0 && this.subscription0.unsubscribe());
        (this.subscription1 && this.subscription1.unsubscribe());
    };
    Wrapper_NgbDatepickerNavigation.prototype.check_date = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_0, currValue))) {
            this._changed = true;
            this.context.date = currValue;
            this._expr_0 = currValue;
        }
    };
    Wrapper_NgbDatepickerNavigation.prototype.check_disabled = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_1, currValue))) {
            this._changed = true;
            this.context.disabled = currValue;
            this._expr_1 = currValue;
        }
    };
    Wrapper_NgbDatepickerNavigation.prototype.check_maxDate = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_2, currValue))) {
            this._changed = true;
            this.context.maxDate = currValue;
            this._expr_2 = currValue;
        }
    };
    Wrapper_NgbDatepickerNavigation.prototype.check_minDate = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_3, currValue))) {
            this._changed = true;
            this.context.minDate = currValue;
            this._expr_3 = currValue;
        }
    };
    Wrapper_NgbDatepickerNavigation.prototype.check_showSelect = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_4, currValue))) {
            this._changed = true;
            this.context.showSelect = currValue;
            this._expr_4 = currValue;
        }
    };
    Wrapper_NgbDatepickerNavigation.prototype.check_showWeekNumbers = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_5, currValue))) {
            this._changed = true;
            this.context.showWeekNumbers = currValue;
            this._expr_5 = currValue;
        }
    };
    Wrapper_NgbDatepickerNavigation.prototype.ngDoCheck = function (view, el, throwOnChange) {
        var changed = this._changed;
        this._changed = false;
        return changed;
    };
    Wrapper_NgbDatepickerNavigation.prototype.checkHost = function (view, componentView, el, throwOnChange) {
    };
    Wrapper_NgbDatepickerNavigation.prototype.handleEvent = function (eventName, $event) {
        var result = true;
        return result;
    };
    Wrapper_NgbDatepickerNavigation.prototype.subscribe = function (view, _eventHandler, emit0, emit1) {
        this._eventHandler = _eventHandler;
        if (emit0) {
            (this.subscription0 = this.context.navigate.subscribe(_eventHandler.bind(view, 'navigate')));
        }
        if (emit1) {
            (this.subscription1 = this.context.select.subscribe(_eventHandler.bind(view, 'select')));
        }
    };
    return Wrapper_NgbDatepickerNavigation;
}());
var renderType_NgbDatepickerNavigation_Host = import3.createRenderComponentType('', 0, import5.ViewEncapsulation.None, [], {});
var View_NgbDatepickerNavigation_Host0 = (function (_super) {
    __extends(View_NgbDatepickerNavigation_Host0, _super);
    function View_NgbDatepickerNavigation_Host0(viewUtils, parentView, parentIndex, parentElement) {
        _super.call(this, View_NgbDatepickerNavigation_Host0, renderType_NgbDatepickerNavigation_Host, import6.ViewType.HOST, viewUtils, parentView, parentIndex, parentElement, import1.ChangeDetectorStatus.CheckAlways);
    }
    View_NgbDatepickerNavigation_Host0.prototype.createInternal = function (rootSelector) {
        this._el_0 = import3.selectOrCreateRenderHostElement(this.renderer, 'ngb-datepicker-navigation', import3.EMPTY_INLINE_ARRAY, rootSelector, null);
        this.compView_0 = new View_NgbDatepickerNavigation0(this.viewUtils, this, 0, this._el_0);
        this._NgbDatepickerNavigation_0_3 = new Wrapper_NgbDatepickerNavigation(this.injectorGet(import8.NgbDatepickerI18n, this.parentIndex), this.injectorGet(import9.NgbCalendar, this.parentIndex));
        this.compView_0.create(this._NgbDatepickerNavigation_0_3.context);
        this.init(this._el_0, (this.renderer.directRenderer ? null : [this._el_0]), null);
        return new import7.ComponentRef_(0, this, this._el_0, this._NgbDatepickerNavigation_0_3.context);
    };
    View_NgbDatepickerNavigation_Host0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import0.NgbDatepickerNavigation) && (0 === requestNodeIndex))) {
            return this._NgbDatepickerNavigation_0_3.context;
        }
        return notFoundResult;
    };
    View_NgbDatepickerNavigation_Host0.prototype.detectChangesInternal = function (throwOnChange) {
        this._NgbDatepickerNavigation_0_3.ngDoCheck(this, this._el_0, throwOnChange);
        this.compView_0.detectChanges(throwOnChange);
    };
    View_NgbDatepickerNavigation_Host0.prototype.destroyInternal = function () {
        this.compView_0.destroy();
        this._NgbDatepickerNavigation_0_3.ngOnDestroy();
    };
    View_NgbDatepickerNavigation_Host0.prototype.visitRootNodesInternal = function (cb, ctx) {
        cb(this._el_0, ctx);
    };
    return View_NgbDatepickerNavigation_Host0;
}(import2.AppView));
export var NgbDatepickerNavigationNgFactory = new import7.ComponentFactory('ngb-datepicker-navigation', View_NgbDatepickerNavigation_Host0, import0.NgbDatepickerNavigation);
var styles_NgbDatepickerNavigation = ['.collapsed[_ngcontent-%COMP%] {\n        margin-bottom: -1.7rem;\n    }'];
var renderType_NgbDatepickerNavigation = import3.createRenderComponentType('', 0, import5.ViewEncapsulation.Emulated, styles_NgbDatepickerNavigation, {});
export var View_NgbDatepickerNavigation0 = (function (_super) {
    __extends(View_NgbDatepickerNavigation0, _super);
    function View_NgbDatepickerNavigation0(viewUtils, parentView, parentIndex, parentElement) {
        _super.call(this, View_NgbDatepickerNavigation0, renderType_NgbDatepickerNavigation, import6.ViewType.COMPONENT, viewUtils, parentView, parentIndex, parentElement, import1.ChangeDetectorStatus.CheckAlways);
        this._expr_25 = import1.UNINITIALIZED;
        this._expr_26 = import1.UNINITIALIZED;
        this._expr_27 = import1.UNINITIALIZED;
    }
    View_NgbDatepickerNavigation0.prototype.createInternal = function (rootSelector) {
        var parentRenderNode = this.renderer.createViewRoot(this.parentElement);
        this._text_0 = this.renderer.createText(parentRenderNode, '\n    ', null);
        this._el_1 = import3.createRenderElement(this.renderer, parentRenderNode, 'table', new import3.InlineArray2(2, 'class', 'w-100'), null);
        this._text_2 = this.renderer.createText(this._el_1, '\n      ', null);
        this._el_3 = import3.createRenderElement(this.renderer, this._el_1, 'tbody', import3.EMPTY_INLINE_ARRAY, null);
        this._el_4 = import3.createRenderElement(this.renderer, this._el_3, 'tr', import3.EMPTY_INLINE_ARRAY, null);
        this._text_5 = this.renderer.createText(this._el_4, '\n        ', null);
        this._el_6 = import3.createRenderElement(this.renderer, this._el_4, 'td', new import3.InlineArray2(2, 'class', 'text-sm-left'), null);
        this._text_7 = this.renderer.createText(this._el_6, '\n          ', null);
        this._el_8 = import3.createRenderElement(this.renderer, this._el_6, 'button', new import3.InlineArray4(4, 'class', 'btn btn-sm btn-secondary btn-inline', 'type', 'button'), null);
        this._text_9 = this.renderer.createText(this._el_8, '<', null);
        this._text_10 = this.renderer.createText(this._el_6, '\n        ', null);
        this._text_11 = this.renderer.createText(this._el_4, '\n        \n        ', null);
        this._anchor_12 = this.renderer.createTemplateAnchor(this._el_4, null);
        this._vc_12 = new import10.ViewContainer(12, 4, this, this._anchor_12);
        this._TemplateRef_12_5 = new import12.TemplateRef_(this, 12, this._anchor_12);
        this._NgIf_12_6 = new import11.Wrapper_NgIf(this._vc_12.vcRef, this._TemplateRef_12_5);
        this._text_13 = this.renderer.createText(this._el_4, '        \n        \n        ', null);
        this._el_14 = import3.createRenderElement(this.renderer, this._el_4, 'td', new import3.InlineArray2(2, 'class', 'text-sm-right'), null);
        this._text_15 = this.renderer.createText(this._el_14, '\n          ', null);
        this._el_16 = import3.createRenderElement(this.renderer, this._el_14, 'button', new import3.InlineArray4(4, 'class', 'next btn btn-sm btn-secondary btn-inline', 'type', 'button'), null);
        this._text_17 = this.renderer.createText(this._el_16, '>', null);
        this._text_18 = this.renderer.createText(this._el_14, '\n        ', null);
        this._text_19 = this.renderer.createText(this._el_4, '\n      ', null);
        this._text_20 = this.renderer.createText(this._el_3, '\n    ', null);
        this._text_21 = this.renderer.createText(parentRenderNode, '\n  ', null);
        var disposable_0 = import3.subscribeToRenderElement(this, this._el_8, new import3.InlineArray2(2, 'click', null), this.eventHandler(this.handleEvent_8));
        var disposable_1 = import3.subscribeToRenderElement(this, this._el_16, new import3.InlineArray2(2, 'click', null), this.eventHandler(this.handleEvent_16));
        this.init(null, (this.renderer.directRenderer ? null : [
            this._text_0,
            this._el_1,
            this._text_2,
            this._el_3,
            this._el_4,
            this._text_5,
            this._el_6,
            this._text_7,
            this._el_8,
            this._text_9,
            this._text_10,
            this._text_11,
            this._anchor_12,
            this._text_13,
            this._el_14,
            this._text_15,
            this._el_16,
            this._text_17,
            this._text_18,
            this._text_19,
            this._text_20,
            this._text_21
        ]), [
            disposable_0,
            disposable_1
        ]);
        return null;
    };
    View_NgbDatepickerNavigation0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import12.TemplateRef) && (12 === requestNodeIndex))) {
            return this._TemplateRef_12_5;
        }
        if (((token === import13.NgIf) && (12 === requestNodeIndex))) {
            return this._NgIf_12_6.context;
        }
        return notFoundResult;
    };
    View_NgbDatepickerNavigation0.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_12_0_0 = this.context.showSelect;
        this._NgIf_12_6.check_ngIf(currVal_12_0_0, throwOnChange, false);
        this._NgIf_12_6.ngDoCheck(this, this._anchor_12, throwOnChange);
        this._vc_12.detectChangesInNestedViews(throwOnChange);
        var currVal_25 = !this.context.showSelect;
        if (import3.checkBinding(throwOnChange, this._expr_25, currVal_25)) {
            this.renderer.setElementClass(this._el_1, 'collapsed', currVal_25);
            this._expr_25 = currVal_25;
        }
        var currVal_26 = this.context.prevDisabled();
        if (import3.checkBinding(throwOnChange, this._expr_26, currVal_26)) {
            this.renderer.setElementProperty(this._el_8, 'disabled', currVal_26);
            this._expr_26 = currVal_26;
        }
        var currVal_27 = this.context.nextDisabled();
        if (import3.checkBinding(throwOnChange, this._expr_27, currVal_27)) {
            this.renderer.setElementProperty(this._el_16, 'disabled', currVal_27);
            this._expr_27 = currVal_27;
        }
    };
    View_NgbDatepickerNavigation0.prototype.destroyInternal = function () {
        this._vc_12.destroyNestedViews();
    };
    View_NgbDatepickerNavigation0.prototype.createEmbeddedViewInternal = function (nodeIndex) {
        if ((nodeIndex == 12)) {
            return new View_NgbDatepickerNavigation1(this.viewUtils, this, 12, this._anchor_12, this._vc_12);
        }
        return null;
    };
    View_NgbDatepickerNavigation0.prototype.handleEvent_8 = function (eventName, $event) {
        this.markPathToRootAsCheckOnce();
        var result = true;
        if ((eventName == 'click')) {
            var pd_sub_0 = (this.context.doNavigate(this.context.navigation.PREV) !== false);
            result = (pd_sub_0 && result);
        }
        return result;
    };
    View_NgbDatepickerNavigation0.prototype.handleEvent_16 = function (eventName, $event) {
        this.markPathToRootAsCheckOnce();
        var result = true;
        if ((eventName == 'click')) {
            var pd_sub_0 = (this.context.doNavigate(this.context.navigation.NEXT) !== false);
            result = (pd_sub_0 && result);
        }
        return result;
    };
    return View_NgbDatepickerNavigation0;
}(import2.AppView));
var View_NgbDatepickerNavigation1 = (function (_super) {
    __extends(View_NgbDatepickerNavigation1, _super);
    function View_NgbDatepickerNavigation1(viewUtils, parentView, parentIndex, parentElement, declaredViewContainer) {
        _super.call(this, View_NgbDatepickerNavigation1, renderType_NgbDatepickerNavigation, import6.ViewType.EMBEDDED, viewUtils, parentView, parentIndex, parentElement, import1.ChangeDetectorStatus.CheckAlways, declaredViewContainer);
    }
    View_NgbDatepickerNavigation1.prototype.createInternal = function (rootSelector) {
        this._el_0 = import3.createRenderElement(this.renderer, null, 'td', import3.EMPTY_INLINE_ARRAY, null);
        this._text_1 = this.renderer.createText(this._el_0, '\n          ', null);
        this._el_2 = import3.createRenderElement(this.renderer, this._el_0, 'ngb-datepicker-navigation-select', import3.EMPTY_INLINE_ARRAY, null);
        this.compView_2 = new import15.View_NgbDatepickerNavigationSelect0(this.viewUtils, this, 2, this._el_2);
        this._NgbDatepickerNavigationSelect_2_3 = new import15.Wrapper_NgbDatepickerNavigationSelect(this.parentView.parentView.injectorGet(import8.NgbDatepickerI18n, this.parentView.parentIndex), this.parentView.parentView.injectorGet(import9.NgbCalendar, this.parentView.parentIndex));
        this._text_3 = this.renderer.createText(null, '\n          ', null);
        this.compView_2.create(this._NgbDatepickerNavigationSelect_2_3.context);
        this._text_4 = this.renderer.createText(this._el_0, '\n        ', null);
        var disposable_0 = import3.subscribeToRenderElement(this, this._el_2, new import3.InlineArray2(2, 'select', null), this.eventHandler(this.handleEvent_2));
        this._NgbDatepickerNavigationSelect_2_3.subscribe(this, this.eventHandler(this.handleEvent_2), true);
        this.init(this._el_0, (this.renderer.directRenderer ? null : [
            this._el_0,
            this._text_1,
            this._el_2,
            this._text_3,
            this._text_4
        ]), [disposable_0]);
        return null;
    };
    View_NgbDatepickerNavigation1.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import14.NgbDatepickerNavigationSelect) && ((2 <= requestNodeIndex) && (requestNodeIndex <= 3)))) {
            return this._NgbDatepickerNavigationSelect_2_3.context;
        }
        return notFoundResult;
    };
    View_NgbDatepickerNavigation1.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_2_0_0 = this.parentView.context.date;
        this._NgbDatepickerNavigationSelect_2_3.check_date(currVal_2_0_0, throwOnChange, false);
        var currVal_2_0_1 = this.parentView.context.disabled;
        this._NgbDatepickerNavigationSelect_2_3.check_disabled(currVal_2_0_1, throwOnChange, false);
        var currVal_2_0_2 = this.parentView.context.maxDate;
        this._NgbDatepickerNavigationSelect_2_3.check_maxDate(currVal_2_0_2, throwOnChange, false);
        var currVal_2_0_3 = this.parentView.context.minDate;
        this._NgbDatepickerNavigationSelect_2_3.check_minDate(currVal_2_0_3, throwOnChange, false);
        this._NgbDatepickerNavigationSelect_2_3.ngDoCheck(this, this._el_2, throwOnChange);
        this.compView_2.detectChanges(throwOnChange);
    };
    View_NgbDatepickerNavigation1.prototype.destroyInternal = function () {
        this.compView_2.destroy();
        this._NgbDatepickerNavigationSelect_2_3.ngOnDestroy();
    };
    View_NgbDatepickerNavigation1.prototype.visitRootNodesInternal = function (cb, ctx) {
        cb(this._el_0, ctx);
    };
    View_NgbDatepickerNavigation1.prototype.handleEvent_2 = function (eventName, $event) {
        this.markPathToRootAsCheckOnce();
        var result = true;
        if ((eventName == 'select')) {
            var pd_sub_0 = (this.parentView.context.selectDate($event) !== false);
            result = (pd_sub_0 && result);
        }
        return result;
    };
    return View_NgbDatepickerNavigation1;
}(import2.AppView));
